### ISSUE TITLE
Add Nix Flake with derivation, shell, service

### ISSUE THREAD

**Issue by jakubgs** [codex-storage/nim-codex#940]
It would be helpful to have a Nix flake for this package. The most basic version would just include a build shell, but a a more advanced version would also include a full derivation and also a NixOS service module.

If possible we should also have CI job for our Nix builds.

**Comment by jakubgs** [comment]
Here is an attempt by Corey to get something working, some parts of the derivation might be useful: [codex-flake.nix.txt](https://github.com/user-attachments/files/17275441/codex-flake.nix.txt)

**Comment by veaceslavdoina** [comment]
@markoburcul, the following lines in [Makefile](https://github.com/codex-storage/nim-codex/pull/954/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52)
```diff
+ CXXFLAGS ?= -std=c++17 -mssse3
+ export CXXFLAGS
```
broke the builds on ARM64 with the error - `c++: error: unrecognized command-line option ‘-mssse3’`

<details>
<summary>more details</summary>

![Image](https://github.com/user-attachments/assets/c21e6b28-5c03-4c07-94c3-83ac5931b245)


```shell
CMake Error at /usr/share/cmake-3.22/Modules/CMakeTestCXXCompiler.cmake:62 (message):
  The C++ compiler

    "/usr/bin/c++"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /opt/nim-codex/vendor/nim-leveldbstatic/build/CMakeFiles/CMakeTmp

    Run Build Command(s):/usr/bin/gmake -f Makefile cmTC_dc503/fast && gmake[1]: Entering directory '/opt/nim-codex/vendor/nim-leveldbstatic/build/CMakeFiles/CMakeTmp'
    /usr/bin/gmake  -f CMakeFiles/cmTC_dc503.dir/build.make CMakeFiles/cmTC_dc503.dir/build
    gmake[2]: Entering directory '/opt/nim-codex/vendor/nim-leveldbstatic/build/CMakeFiles/CMakeTmp'
    Building CXX object CMakeFiles/cmTC_dc503.dir/testCXXCompiler.cxx.o
    /usr/bin/c++   -std=c++17 -mssse3  -o CMakeFiles/cmTC_dc503.dir/testCXXCompiler.cxx.o -c /opt/nim-codex/vendor/nim-leveldbstatic/build/CMakeFiles/CMakeTmp/testCXXCompiler.cxx
    c++: error: unrecognized command-line option ‘-mssse3’
    gmake[2]: *** [CMakeFiles/cmTC_dc503.dir/build.make:78: CMakeFiles/cmTC_dc503.dir/testCXXCompiler.cxx.o] Error 1
    gmake[2]: Leaving directory '/opt/nim-codex/vendor/nim-leveldbstatic/build/CMakeFiles/CMakeTmp'
    gmake[1]: *** [Makefile:127: cmTC_dc503/fast] Error 2
    gmake[1]: Leaving directory '/opt/nim-codex/vendor/nim-leveldbstatic/build/CMakeFiles/CMakeTmp'





  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:7 (project)
```
</details>

**Comment by veaceslavdoina** [comment]
`mssse3` are x86 specific and we should not use them for arm

- [3.19.5 ARM Options](https://gcc.gnu.org/onlinedocs/gcc/ARM-Options.html#ARM-Options-1)
- [3.19.54 x86 Options](https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html)
> ‘sse’
Use scalar floating-point instructions present in the SSE instruction set. This instruction set is supported by Pentium III and newer chips, and in the AMD line by Athlon-4, Athlon XP and Athlon MP chips. The earlier version of the SSE instruction set supports only single-precision arithmetic, thus the double and extended-precision arithmetic are still done using 387. A later version, present only in Pentium 4 and AMD x86-64 chips, supports double-precision arithmetic too.

**Comment by jakubgs** [comment]
Would be good to update the `nix/README.md` with these service setup instructions too. But other than that good work.

**Comment by markoburcul** [comment]
The pipeline for the nim codex is now available on our [Jenkins instance](https://ci.status.im/job/nim-codex/job/systems/job/linux/job/x86_64/job/nix/). I'm gonna close this issue since all of the components have been delivered including:
- nix shell
- nix derivation
- nix service
- Jenkins pipeline

**Comment by jakubgs** [comment]
Yes, very good work. We can now use what we learned here and apply it to `nim-waku`.
